[metis] Use Debian source archive as the upstream server is down

p12tic opened the pull request:
http://glaros.dtc.umn.edu/gkhome/metis/metis/download has been down since at least couple of weeks ago. This breaks all CI on Windows as the source package can't be downloaded. We can switch to the source package hosted to Debian instead until the upstream recovers. The Debian source package is different as it removed manual PDF document due to missing source of the document.

fabiencastan commented on ports/metis/portfile.cmake:
```diff
@@ -4,9 +4,9 @@ set(OPTIONS -DSHARED=OFF)
 vcpkg_download_distfile(ARCHIVE
-    URLS "http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/metis-${METIS_VERSION}.tar.gz"
-    FILENAME "metis-${METIS_VERSION}.tar.gz"
-    SHA512 deea47749d13bd06fbeaf98a53c6c0b61603ddc17a43dae81d72c8015576f6495fd83c11b0ef68d024879ed5415c14ebdbd87ce49c181bdac680573bea8bdb25
+    URLS "https://mirrors.edge.kernel.org/debian/pool/main/m/metis/metis_5.1.0.dfsg.orig.tar.xz"
+    FILENAME "metis_5.1.0.tar.xz"
+    SHA512 77d662dfade6131e17b934ba1b7f02e41dec701cfb0e0f0afce420e3a02489611200e323e288d88cd8de004b4d491c6a115ce0a9f9a0df1caba9f5682c67f2f0
```
It would be good to sill use "${METIS_VERSION}" and keep a comment to explain that we use a more reliable mirror than the original repository with the link to it.

BillyONeal committed:
[gklib,metis,parmetis] Update sources for metis et al. to GitHub (#26020)

* Remove bogus arguments in the `vcpkg create` template.

* [gklib] Add new port in support of metis and parmetis.

New port checklist:
* Review the code :)
* Check the name against https://repology.org/ There are 2 packages, both of which seem to be the same thing
* Check the name against Bing/Google. It's the first entry.
* Check the source code for optional `find_package`s There are none.
* Check that the versioning scheme and license match what upstream says They have none.
* Check that the source code comes from the upstream project's authoritative source. We did.
* Check that the generated usage is accurate.
* Check that the reported license is accurate. We hope so.

* [metis] Update source to github.

Resolves https://github.com/microsoft/vcpkg/issues/25764

Competes with https://github.com/microsoft/vcpkg/pull/25953

* [parmetis] Update source to github.

* Add licenses.

* Fix GKlib case.

* Fix metis install.

p12tic commented:
Looks like upstream fixed this issue in a different way: https://github.com/microsoft/vcpkg/pull/26020. I think it makes sense to cherry-pick that commit instead.
